packages/sdk40/src/structs.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { cosmosField, cosmosMessage, Registry } from "@cosmjs/proto-signing";
+import { Message } from "protobufjs";
+
+export const defaultRegistry = new Registry();
```

Comment:
I think we don't need the registry at all for those use cases because we know what type is expected. In #352 I use `BaseAccount` without a registry.

Reply:
This may be ModuleAccount, VestingAccount or ContinuousVestingAccount.

Forcing it to BaseAccount may fail on some chains

Reply:
I see. Happy to generalize this. But with the current registry aproach we can only encode in a typesafe way, not decode. We can extend this switch to more account types:

```ts
const { typeUrl, value } = decodeAny(responseData);
    switch (typeUrl) {
      case "/cosmos.auth.BaseAccount": {
        const { account_number, sequence } = BaseAccount.decode(value);
        assert(account_number !== undefined);
        assert(sequence !== undefined);
        return {
          accountNumber: uint64FromProto(account_number).toNumber(),
          sequence: uint64FromProto(sequence).toNumber(),
        };
      }
      default:
        throw new Error(`Unsupported type: ${typeUrl}`);
    }
```

Reply:
I should add some other types of accounts to genesis...

Reply:
```
Usage:
  simd add-genesis-account [address_or_key_name] [coin][,[coin]] [flags]

Flags:
      --vesting-amount string    amount of coins for vesting accounts
      --vesting-end-time int     schedule end time (unix epoch) for vesting accounts
      --vesting-start-time int   schedule start time (unix epoch) for vesting accounts
```